eth/api_backend.go

```diff
@@ -67,11 +67,18 @@ func (b *EthAPIBackend) HeaderByNumber(ctx context.Context, number rpc.BlockNumb
 	// Pending block is only known by the miner
 	if number == rpc.PendingBlockNumber {
 		block := b.eth.miner.PendingBlock()
-		return block.Header(), nil
+		if block != nil {
```

Comment:
The pending block - if I understand it correctly - can only be empty on startup, but will then be filled and will always have value afterwards.